nixos/modules/services/misc/stash-bin.nix

```diff
@@ -0,0 +1,117 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.stash;
+  opt = options.services.stash;
+
+in
+{
+
+  options.services.stash = {
+    enable = mkEnableOption (mdDoc "Stash");
+
+    user = mkOption {
+      type = types.str;
+      default = "stash";
+      description = mdDoc "User account under which Stash runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "stash";
+      description = mdDoc "Group under which Stash runs.";
+    };
+
+    host = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
+      description = mdDoc "The ip address for the host that stash is listening to.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 9999;
+      description = mdDoc "The port that stash serves to.";
+    };
+
+    externalHost = mkOption {
+      type = types.str;
+      default = "";
+      description = mdDoc "Needed in some cases when you use a reverse proxy.";
+    };
+
+    dataDir = mkOption {
+      type = types.path;
+      default = "/var/lib/stash";
+      description = mdDoc "Path where to store data files.";
+    };
+
+    package = mkPackageOptionMD pkgs "stash" {};
+};
+
+  config = mkIf cfg.enable (
+    let
+      stashPython = pkgs.python3.withPackages (ps: with ps; [
+        cloudscraper
+        configparser
+        progressbar
+        requests
+      ]);
+      stashPackages = with pkgs; [
+        stashPython
+        bashInteractive
+        openssl
+        sqlite
+        ffmpeg
+      ];
+    in
+    {
+      systemd.services.stash = {
+        description = mdDoc "Stash daemon";
+        after = [ "network.target" ];
+        wantedBy = [ "multi-user.target" ];
+        environment = {
+          HOME = "%S/stash";
```

Comment:
I'm guessing you can set these:
https://github.com/stashapp/stash/blob/f69bd8a94ffbb4a796996da448fcbefdac6685b7/internal/manager/config/init.go#L140-L143
instead of relying on HOME